File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyidconnectionsconnectioniddatatransferstransferid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCompaniesCompanyIdConnectionsConnectionIdDataTransfersTransferIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=connectionId" })
  connectionId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=transferId" })
  transferId: string;
}


export class GetCompaniesCompanyIdConnectionsConnectionIdDataTransfersTransferIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class GetCompaniesCompanyIdConnectionsConnectionIdDataTransfersTransferIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdConnectionsConnectionIdDataTransfersTransferIdPathParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdConnectionsConnectionIdDataTransfersTransferIdSecurity;
}


export class GetCompaniesCompanyIdConnectionsConnectionIdDataTransfersTransferIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataContractsDatasetsTransfer?: shared.CodatDataContractsDatasetsTransfer;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/getcompaniescompanyidconnectionsconnectioniddatatransferstransferid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDConnectionsConnectionIDDataTransfersTransferIDPathParams struct {
	CompanyID    string `pathParam:"style=simple,explode=false,name=companyId"`
	ConnectionID string `pathParam:"style=simple,explode=false,name=connectionId"`
	TransferID   string `pathParam:"style=simple,explode=false,name=transferId"`
}

type GetCompaniesCompanyIDConnectionsConnectionIDDataTransfersTransferIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDConnectionsConnectionIDDataTransfersTransferIDRequest struct {
	PathParams GetCompaniesCompanyIDConnectionsConnectionIDDataTransfersTransferIDPathParams
	Security   GetCompaniesCompanyIDConnectionsConnectionIDDataTransfersTransferIDSecurity
}

type GetCompaniesCompanyIDConnectionsConnectionIDDataTransfersTransferIDResponse struct {
	CodatDataContractsDatasetsTransfer *shared.CodatDataContractsDatasetsTransfer
	ContentType                        string
	StatusCode                         int
}
